airbyte-workers/src/test/java/io/airbyte/workers/general/DefaultCheckConnectionWorkerTest.java

```diff
@@ -88,9 +105,65 @@ void testEnums() {
 
   @Test
   void testSuccessfulConnection() throws WorkerException {
-    final DefaultCheckConnectionWorker worker = new DefaultCheckConnectionWorker(integrationLauncher, successStreamFactory);
+    final DefaultCheckConnectionWorker worker = new DefaultCheckConnectionWorker(integrationLauncher, connectorConfigUpdater, successStreamFactory);
+    final ConnectorJobOutput output = worker.run(input, jobRoot);
+
+    verifyNoInteractions(connectorConfigUpdater);
+
+    assertEquals(output.getOutputType(), OutputType.CHECK_CONNECTION);
+    assertNull(output.getFailureReason());
+
+    final StandardCheckConnectionOutput checkOutput = output.getCheckConnection();
+    assertEquals(Status.SUCCEEDED, checkOutput.getStatus());
+    assertNull(checkOutput.getMessage());
+  }
+
+  @Test
+  void testCheckConnectionWithConfigUpdateSource() throws WorkerException {
+    final DefaultCheckConnectionWorker worker =
+        new DefaultCheckConnectionWorker(integrationLauncher, connectorConfigUpdater, configMessageStreamFactory);
     final ConnectorJobOutput output = worker.run(input, jobRoot);
 
+    verify(connectorConfigUpdater).updateSource(ACTOR_ID, CONNECTOR_CONFIG);
+    verifyNoMoreInteractions(connectorConfigUpdater);
```

Comment:
I'm having trouble reading this... does this mean that you are asserting that attempting to update the config again does nothing because the values already have been updated within the check?

Reply:
`verifyNoMoreInteractions` just checks that we're not making any other calls to the connectorConfigUpdater. Since the worker can choose to use updateSource or updateDestination based on the type of actor I just wanted to verify that the other method isn't getting called.